### ISSUE TITLE
How to add FNC1 to Value

### ISSUE THREAD

**Issue by LordPinhead** [barnhill/barcodelib#29]
Hi Brad,

i really like your Barcode Library, from all Libs we try´d, it works like a charm.
The only thing i need is what i have to add as an FNC1 Char.

In the Code128.cs Symbology file, i saw FNC1 is Char 200, so i tested it but it does not work for my Scanner when i try the following code:

```
BarcodeLib.Barcode b = new BarcodeLib.Barcode();
b.IncludeLabel = true;
var value = String.Format("10L17146133619{0}0104000358091121", Convert.ToChar(200).ToString());
b.Encode(BarcodeLib.TYPE.CODE128, value, 10*65, 200);
b.SaveImage(sfd.FileName, savetype);
```

Is my Test Scanner (some Jartech) not configured properly or did i something wrong in my Barcode?

The value is legit so far and should read like:
Field: 10 Value: L17146133619 Length: 12 byte
Field: 01 Value: 04000358091121 Length: 14

The last field is OK, because it is static length, but field 10 is variable and should be terminated by the FNC1.

Maybe you have 2 Minutes to explain it.

And btw. when i activate the label text under it, the FNC1 char is printed in the label, maybe it is a bug or my fault, so i just wait what your answer is.

Regards
Thomas

**Comment by LordPinhead** [comment]
1) There is a Bug in Code128.cs when you want FNC1 Divider in the middle of your value, it will never find it and in FindStartorCodeCharacter() it looks like to me like a Copy-and-Paste Bug:

```
if (s.Length > 1 && (Char.IsNumber(s[0]) || s[0] == Convert.ToChar(200)) && (Char.IsNumber(s[1]) || s[0] == Convert.ToChar(200)))
```

Maybe it should be:

```
if (s.Length > 1 && (Char.IsNumber(s[0]) || s[0] == Convert.ToChar(200)) && (Char.IsNumber(s[1]) || s[1] == Convert.ToChar(200)))
```

The last ArraySelection is of by 1.

When i change it and build Barcodes, the scanner finds my FNC1 Code.

2) Printing a Label with 128: The FNC1 Divider is still in cleartext in the label line as "È", i try to fix it today.

3) Maybe it would be a good idea to use non printable chars like 29 (Group Separator)? Any Opinion an that?

**Comment by barnhill** [comment]
Good catch on the array index being off.  Ill fix that now.

**Comment by barnhill** [comment]
Has this issue been fixed with https://github.com/barnhill/barcodelib/commit/6c9b77bf1ef29fae8075145e25edd89273e63b92

I'll close this issue if so.

**Comment by nabber** [comment]
I'm doing some testing and I also need the FNC1 character.  How can I add this character into my "Value to Encode" field?  

![image](https://user-images.githubusercontent.com/3247525/30561042-632a7c82-9c7f-11e7-9dcc-27fbf15dad89.png)

**Comment by LordPinhead** [comment]
In theory, you could use ALT+200 - i did not use the gui, i used Unittests for that.

**Comment by barnhill** [comment]
The included sample application isnt meant to cover all cases.  It was a quick and pretty dirty way to just showcase most of the libraries funcationality.  Its not production ready code (the sample app) or anything, just a quick POC.

**Comment by barnhill** [comment]
@LordPinhead  did you see my comment above about asking if this has been fixed with 6c9b77b

**Comment by LordPinhead** [comment]
@Barnhill yes it is solved.
